tfgrid3/staticwebsite/zinit/clone.yaml

```diff
@@ -0,0 +1,2 @@
+exec:  bash -c 'git clone $GITHUB_URL website'
```

Comment:
this line also need to be changed to run with exec only and without bash .

Reply:
When I run it with exec only, The process fails as it views the ```$GITHUB_URL``` as a string not as env var
![2024-05-29_16-13](https://github.com/threefoldtech/tf-images/assets/66883096/38f6b4a9-1671-482b-b9b7-af3d30954db2)

Reply:
so , let's keep it as it now .

Reply:
Let me explain again because I did explain this multiple times before.

in the yaml file when you do 
```
exec: some command with argument
```
This command is `executed` without a shell. it basically started with one of `fork/execv` syscalls. The ENV vars WILL be available for the process to read pragmatically (for example, using `getenv` in C)

This also mean if I did something like `exec: echo $VAR_NAME` what happens is the echo command is started with one argument that is literally `$VAR_NAME`. There is no env var interpretation here. 

The reason that there is NO env var interpretation is that this functionality is actually a feature of a `shell` like `bash`, `sh`, or `zsh`

When you type something like this:
```
echo $VAR_NAME
```
In your shell, it's the SHELL (bash in most cases) that actually interpret the $VAR_NAME to actual value (from the env vars of the shell) and then does the fork/exec and pass the value as the first argument to the echo command. That is one of the shell scripting features (similar to if statements, for loops, etc...)

So, What Mayar is trying to do is then pretty normal. She doesn't want to start a git command, she is starting a `bash script` that needs to does something first before starting the git command, and that thing is evaluating the value of the GITHUB_URL 

The only suggestion I can give here is to do:
```
exec: bash -c 'exec git clone $GITHUB_URL website`
```
> I added an exec before the `git` command

This changes a little bit how this script reads. First the bash will replace GITHUB_URL with value set in the env var. Then instead of doing `fork/exec` it will instead do `exec` which basically replaces the bash process with git and then your process tree will look like 
```
zinit
   -> git
```
Instead of 
```
zinit
  -> bash
     -> git
```